modules/yugabytedb/src/main/java/org/testcontainers/containers/delegate/YugabyteDBYCQLDelegate.java

```diff
@@ -33,9 +33,20 @@ public void execute(
         boolean continueOnError,
         boolean ignoreFailedDrops
     ) {
+        final String containerInterfaceIP = container
+            .getContainerInfo()
+            .getNetworkSettings()
+            .getNetworks()
+            .entrySet()
+            .stream()
+            .findFirst()
+            .get()
+            .getValue()
+            .getIpAddress();
         try {
             ExecResult result = container.execInContainer(
                 BIN_PATH,
+                containerInterfaceIP,
```

Comment:
Why is this change needed as part of the PR? Why was it no necessary before, but it is necessary now?

Reply:
The change is to get the correct IP on which the server listens for all versions of the database.

Reply:
Can you elaborate a little bit more, please? By default, there is one YB instance running in the container, right? So, ycqlsh uses 127.0.0.1 as a host by default which in this case since client and server are in the same instance it works. Is it not that the default behavior? or are we missing something in the YB world?

Reply:
Yes @eddumelendez the default behaviour you described is correct for the current YugabyteDB version (2.14).
However, for some newer versions the server binds to the network IP-address after resolving 0.0.0.0 by default.

**This change in the PR ensures that we get the right address to connect to, irrespective of the version of YugabyteDB**

Reply:
Is there a image version that can produce the issue? If so, then we can add a test for it. Just want to make sure we are covering correctly the issues. It is totally fine to test again different image versions if those contain changes that can break existing implementation. But, we should take into account that new changes should be backward compatible.

Also, please run `./gradlew spotlessApply` before pushing the changes.

Reply:
my suggestion would be to revert those changes and discuss about it later in order to do not block the original change. WDYT?

Reply:
The issue can be produced with `2.18` version of YugabyteDB. For example: The following test would fail with 2.18 without above changes (Will add this test case for 2.18) : 
```
public void testSmoke() {
        try (
            // creatingYCQLContainer {
            final YugabyteDBYCQLContainer ycqlContainer = new YugabyteDBYCQLContainer(
                "yugabytedb/yugabyte:2.18.3.0-b75"
            )
            // }
        ) {
            // startingYCQLContainer {
            ycqlContainer.start();
            // }
            assertThat(performQuery(ycqlContainer, "SELECT release_version FROM system.local").wasApplied())
                .as("A sample test query succeeds")
                .isTrue();
        }
}
```
The users would most likely be using testcontainers  with latest YugabyteDB versions than 2.14. 
>  But, we should take into account that new changes should be backward compatible.

**I would like to highlight that these changes would work with all the versions of YugabyteDB, 2.14 included.**

Reply:
thanks! I think adding the issue will cover the scenario and remind us about why that change was introduced

Reply:
@eddumelendez, I have added the test case and also have taken care of `./gradlew spotlessApply`